simulation/matching/matching_simulation_test.go

```diff
@@ -213,8 +213,8 @@ func (s *MatchingSimulationSuite) TestMatchingSimulation() {
 			TaskList:     &types.TaskList{Name: tasklist, Kind: types.TaskListKindNormal.Ptr()},
 			TaskListType: types.TaskListTypeDecision.Ptr(),
 			PartitionConfig: &types.TaskListPartitionConfig{
-				NumReadPartitions:  int32(getPartitions(s.TestClusterConfig.MatchingConfig.SimulationConfig.TaskListReadPartitions)),
-				NumWritePartitions: int32(getPartitions(s.TestClusterConfig.MatchingConfig.SimulationConfig.TaskListWritePartitions)),
+				ReadPartitions:  getPartitions(s.TestClusterConfig.MatchingConfig.SimulationConfig.TaskListReadPartitions),
```

Comment:
we should probably add at least "default" case to the CI to catch these in the future (can also validate latencies are within some range etc.)